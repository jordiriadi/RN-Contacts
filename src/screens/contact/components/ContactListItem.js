import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import { Avatar } from 'react-native-elements';
import Styles from '../../../utils/Styles';

const ContactListItem = (props) => {
  const {contact, onPress} = props;
 
  return (
    <TouchableOpacity key={contact.id} style={styles.container} onPress={() => onPress(contact)}>
      <View style={styles.avatarContainer}>
        <Avatar
          rounded
          size="medium"
          source={{
            uri:contact.photo
          }}
          title={contact.initial}
        />
      </View>
      <View style={styles.textContainer}>
        <Text style={Styles.DefaultText}>
          {contact.name}
        </Text>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: Styles.DefaultPadding,
    flexDirection: 'row',
  },
  avatarContainer: {
    flex: 1,
    alignItems: 'center',
  },
  textContainer: {
    flex: 5,
    justifyContent: 'center',
    borderBottomColor: Styles.PrimaryColor,
    borderBottomWidth: 1
  }
});

ContactListItem.propTypes = {
  contact: PropTypes.object,
  onPress: PropTypes.func,
}


export default ContactListItem;