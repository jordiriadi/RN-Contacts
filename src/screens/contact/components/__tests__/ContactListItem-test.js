import React from 'react';
import { shallow } from 'enzyme';
import ContactListItem from '../ContactListItem';

describe.only('Testing Contact List Item component', () => {
  const firstContact = {
    "age": "20",
    "firstName": "Luke",
    "id": "b3abd640-c92b-11e8-b02f-cbfa15db428b",
    "initial": "LS", 
    "lastName": "Skywalker", 
    "name": "Luke Skywalker", 
    "photo": "N/A"
  }
  const secondContact = {
    "age": "33",
    "firstName": "LukeTheSecond",
    "id": "b3abd640-b22g-11e8-b02f-cbfa15db428b",
    "initial": "LS", 
    "lastName": "Skywalker", 
    "name": "LukeTheSecond Skywalker", 
    "photo": "N/A"
  }

  it('renders as expected', () => {
    const wrapper = shallow(
      <ContactListItem contact={firstContact} />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ secondContact });
    expect(wrapper).toMatchSnapshot();
  });
});