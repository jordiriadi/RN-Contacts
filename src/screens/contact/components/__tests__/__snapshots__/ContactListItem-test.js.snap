// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Contact List Item component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing Contact List Item component renders as expected 2`] = `ShallowWrapper {}`;
